datafusion/common/src/scalar.rs

```diff
@@ -763,6 +763,42 @@ impl ScalarValue {
         }
     }
 
+    // ListArray compatible version of new_primitive
+    pub fn new_primitives<T: ArrowPrimitiveType>(
+        values: Vec<Option<T::Native>>,
+        d: &DataType,
+    ) -> Result<Self> {
+        if values.is_empty() {
+            return d.try_into();
+        }
+
+        // We need to convert it to ScalarValue::Primitive (Int64(0)) instead of ScalarValue::List (List([PrimitiveArray<Int64> [0]]))
```

Comment:
As a suggestion for future PRs, I think comments are best when they describe the *why* of the code (rather than the what) -- in this case I think the why is because this function is doing something very special depending on what mode the accumulator is in.

Reply:
Nice advice. I will keep this in mind